pallets/clusters/src/lib.rs

```diff
@@ -0,0 +1,476 @@
+			}
+
+			Self::deposit_event(Event::RoleDeleted { role_hash });
+			log::trace!("Role deleted: {:?}", role_hash);
+
+			Ok(())
+		}
+
+		/// Create a **Member**, give it a set of existing **Role** and assign it to an existing **Cluster**.
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		pub fn add_member(
+			origin: OriginFor<T>,
+			cluster_id: Hash128,
+			roles: Vec<Vec<u8>>,
+			member: T::AccountId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			// Check that the cluster exists and the caller is the owner of the cluster
+			let cluster = <Clusters<T>>::get(cluster_id).ok_or(Error::<T>::ClusterNotFound)?;
+			ensure!(who == cluster.owner, Error::<T>::NoPermission);
+
+			// Check that the cluster does not already contain the member
+			ensure!(!cluster.members.contains(&member), Error::<T>::MemberExists);
+
+			// Check that the roles for the member already exists in the cluster
+			let roles_in_cluster: Vec<Vec<u8>> =
+				cluster.roles.iter().map(|role| role.name.clone()).collect();
+			for role in &roles {
+				ensure!(roles_in_cluster.contains(&role), Error::<T>::RoleNotFound);
+			}
+
+			// write
+			<Clusters<T>>::mutate(&cluster_id, |cluster| {
+				let cluster = cluster.as_mut().unwrap();
+				// Add member into the cluster
+				cluster.members.push(member.clone());
+
+				// Associate the member with its roles in the cluster
+				for role in roles.clone() {
+					let index = cluster.roles.iter().position(|x| x.name == role);
+					if let Some(index) = index {
+						let role =
+							cluster.roles.get(index).ok_or(Error::<T>::SystematicFailure).unwrap();
+						let mut role_members = role.clone().members;
+						role_members.push(member.clone());
+					}
+				}
+			});
+
+			Ok(())
+		}
+
+		/// Delete a **Member** from an existing **Cluster**.
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		pub fn delete_member(
+			origin: OriginFor<T>,
+			cluster_id: Hash128,
+			member: T::AccountId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let cluster = <Clusters<T>>::get(cluster_id).ok_or(Error::<T>::ClusterNotFound)?;
+			ensure!(who == cluster.owner, Error::<T>::NoPermission);
+			// Check that the member is in the cluster
+			ensure!(cluster.members.contains(&member), Error::<T>::MemberNotFound);
+
+			// write
+			// Delete member from Cluster
+			<Clusters<T>>::mutate(&cluster_id, |cluster| {
+				let cluster = cluster.as_mut().unwrap();
+				let index = cluster.members.iter().position(|x| x == &member);
+				if let Some(index) = index {
+					cluster.members.remove(index);
+				}
+			});
+
+			Ok(())
+		}
+
+		/// Allow another Account ID `proxy_account` to be used as a proxy for the Account ID of the cluster
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		pub fn add_proxy_account(
+			origin: OriginFor<T>,
+			cluster_id: Hash128,
+			proxy_account: T::AccountId,
```

Comment:
I don't get it, I'd rather just create a proxy here, not add the delegate directly (not the proxy in fact)

Reply:
Proxy now is created inside the function [create_cluster](https://github.com/fragcolor-xyz/clamor/pull/224/files#diff-6804b4445503eaa0a518d6cabdb56e2901be2c3b0c19ba2b7b4a5e4becdad173R233) like so:
```rust
// create a pure proxy
pallet_proxy::Pallet::<T>::create_pure(
	origin.clone(),
	T::ProxyType::default(),
	T::BlockNumber::zero(),
	0,
)?;
```